tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,153 @@
+    LOGGER.debug("Starting test number 1 - agent should be online.")
+
+    result_future_online = ResultFuture()
+    start_time_immediately = time.time()
+    check_thread_online = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_online,
+            "Checking if agent is online",
+            DEFAULT_TIMEOUT,
+            0,
+        ),
+    )
+    check_thread_online.start()
+    check_thread_online.join()
+    elapsed_time = time.time() - start_time_immediately
+    assert (
+        elapsed_time < 1 and result_future_online.result
+    ), "Expected less than 1 second and the agent to be online"
+
+    # Test 2: Stop agent and verify --wait returns 1 after wait time is expired.
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Starting test number 2 - agent should remain offline.")
+
+    result_future_offline = ResultFuture()
+    start_time_wait_5_sec = time.time()
+    check_thread_offline = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_offline,
+            "Checking if agent remains offline",
+            SHORT_TIMEOUT,
+            1,
+        ),
+    )
+    check_thread_offline.start()
+    check_thread_offline.join()
+    elapsed_time_5_sec = time.time() - start_time_wait_5_sec
+    assert 5 < elapsed_time_5_sec < 6, "--wait returned offline after 5 seconds"
+    assert not result_future_offline.result, "Expected agent to remain offline"
+
+    # Test 3: Stop agent, run 'bluechi-is-online agent --wait', start agent and verify --wait returns 0 before the
+    # wait time expires
+    LOGGER.debug(
+        "Starting test number 3, stopping agent to ensure it is inactive before starting `bluechi-is-online` "
+        "thread."
+    )
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Agent confirmed inactive after stopping.")
+
+    result_future_wait = ResultFuture()
+    start_time = time.time()
+    LOGGER.debug("Starting `bluechi-is-online` thread with wait time of 20 seconds.")
+    check_thread_wait = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_wait,
+            "Checking if agent comes online",
+            LONG_TIMEOUT,
+            0,
+        ),
+    )
+    check_thread_wait.start()
+    # Restart the agent after a delay to simulate it coming online during the --wait period
+    time.sleep(SLEEP_DURATION)
+    node_foo.systemctl.start_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "active")
+    LOGGER.debug("Agent confirmed active after starting.")
+
+    # Wait for the thread to complete
+    check_thread_wait.join()
```

Comment:
If the `check_thread_wait` thread hangs - i.e. `bluechi-is-online` keeps running - we'd never stop here. There is already a timeout for the test runs in place, but it is quite high. Since the agent got started the lines before, we can expect the `check_thread_wait.join()` to exit quite soon - so adding a safeguard with ~2s is appropriate:
```python
with Timeout(2, "Thread didn't stop within 2s of starting bluechi-agent"):
    check_thread_wait.join()
```